README.md
### A Preview of Quick Box
[![Quick Box v1.6.5 Dashboard](https://raw.githubusercontent.com/JMSDOnline/quick-box-update/master/assets/quickbox-dasboard-youtube-preview.png)](http://www.youtube.com/watch?v=Nui3Gicc0mI)

## How to install
> This script is valid for current Quick Box installs only. Please visit the Quick Box repo for the full installation kit.

---

### Update for version 1.3 on Ubuntu 14.04, 15.04, and 15.10

**Run the following command to grab our prep script and setup for install ...**
```
curl -LO https://raw.githubusercontent.com/JMSDOnline/quick-box-update/master/update-1-3.sh

```
**... Finally, make the script executable and run to produce the updates ...**
```
chmod +x update-1-3.sh
./update-1-3.sh

```

---

####REMEMBER: You must be logged in as root to run this update.


v1-3/changelog.md
###Changelog for version 1.3 of Quick Box

  * .bashrc - The bashrc file is located within your /root/ directory. This file is the once responsible for many of the commands that you may put to use on your Quick Box seedbox install. Of all these commands, there was but one that was presenting a problem in v1.2. That was the 'changeUserpass' command. 
  The Problem   : Changing a users password would create complications for the user to login afterwards. This was due to the user having a duplicated set of username and password hashes within the /etc/htpasswd file.
  The Solution  : I have written the proper sed command to be fired at user password change. This sed command will remove the last know users password and hash entry from the file before echoing in the new username and password hash.

  * LShell - LShell -or- Limited Shell is a user environment written in python that allows us to restrict a users shell environment to certain commands that we approve. 
  The problem   : Aditional added users could break out of their /home/${username} directories by connectiong to VSFTPd on port 4747 (SFTP). This would give the additional user the ability to browse the entire server.
  The Solution  : Set SFTP options to '0' in the lshell.conf file. This file is located at /etc/lshell.conf if you would like to take a look at it's settings.

  * Plex Public Access  : This one is exciting as it adds in the ability to simply enable (switch on) and disable (switch off) Plex Access WITHOUT the need for creating a tunnel on your local computer. Understandably, for some this is too much of a hassle to enjoy Plex as they prefer to simply watch TV and Movies directly from their server (permitting their network and hardware is suitable). You may opt to enable it when you are wanting to watch it (Public Access - no login required) and disable it when you want to secure your Plex Media Server (Deletes a custom plex.conf file in the apache2/site-enabled directory, thus hiding Plex from Public Access)

  * Dashboard index.php update - To aid in this updated function of Plex, there were a few minor changes that needed to be made to the index file as it uses custom shell hooks via php to make this possible. Is it safe? I wouldn't be doing this if it wasn't!

  Here's a question --- "I have plex from your version 1.0 installed. I think that version was built from source. Your update looks like it adds a Debian ppa to the sources.list and runs an install. Will this affect my current Plex?"

  Here's an answer --- "No, the built-from-source Plex will stay as it is. Since PlexMediaServer is already installed (and more than likely out-of-date) the ppa will simply pickup that Plex is already installed and rather than attempting to install, it will merely run an update... thus making future updates a breeze (ex: apt-get install --only-upgrade plexmediaserver"

v1-6/quickbox-update-v1-6.sh
#!/bin/bash
#
# [Quick Box Update v1.6 Script]
#
# GitHub:   https://github.com/JMSDOnline/quick-box
# Author:   Jason Matthews
# URL:      https://jmsolodesigns.com/code-projects/quick-box/seedbox-installer
#
# 
#################################################################################
HOSTNAME1=$(hostname -s);
VERSION="1.6"
OUTTO="/root/quick-box-update-v1-6.log"
#################################################################################
#Script Console Colors
black=$(tput setaf 0); red=$(tput setaf 1); green=$(tput setaf 2); yellow=$(tput setaf 3); 
blue=$(tput setaf 4); magenta=$(tput setaf 5); cyan=$(tput setaf 6); white=$(tput setaf 7); 
on_red=$(tput setab 1); on_green=$(tput setab 2); on_yellow=$(tput setab 3); on_blue=$(tput setab 4); 
on_magenta=$(tput setab 5); on_cyan=$(tput setab 6); on_white=$(tput setab 7); bold=$(tput bold); 
dim=$(tput dim); underline=$(tput smul); reset_underline=$(tput rmul); standout=$(tput smso); 
reset_standout=$(tput rmso); normal=$(tput sgr0); alert=${white}${on_red}; title=${standout}; 
sub_title=${bold}${yellow}; repo_title=${black}${on_green};
#################################################################################

function _string() { perl -le 'print map {(a..z,A..Z,0..9)[rand 62] } 0..pop' 15 ; }

# Download packaged version 1.6 Quick Box
#curl -LO https://raw.githubusercontent.com/JMSDOnline/quick-box-update/master/v1-6/changelog.md >>"${OUTTO}" 2>&1

# Ask if user would like to update Plex
function _intro() {
  echo
  echo
  echo "${title} Welcome to the Quick Box update kit! ${normal}"
  echo " version: ${VERSION}"
  echo
  echo "The version 1.6 update caters to fixing a few minor (yet annoying) bugs"
  echo "that were occuring on multi-user intended environments. Additonal enhancements"
  echo "such as adding additional built-in functions for tasks such as upgrading"
  echo "included multi-media packages such as BTSync."
  echo
  echo "Updates will run silently. You can review the update log"
  echo "at ${OUTTO}."
  echo
  echo -n "${bold}${yellow}Are you ready to run the v1.6 update?${normal} (${bold}${green}Y${normal}/n): "
  read responce
  case $responce in
    [yY] | [yY][Ee][Ss] | "" ) update=yes ;;
    [nN] | [nN][Oo] ) update=no ;;
  esac
}

clear

function _bashrc() {
  curl -LO https://raw.githubusercontent.com/JMSDOnline/quick-box-update/master/v1-6/.bashrc >>"${OUTTO}" 2>&1
  curl -o /usr/bin/quickbox https://raw.githubusercontent.com/JMSDOnline/quick-box-update/master/v1-6/quickbox >>"${OUTTO}" 2>&1
  chmod +x /usr/bin/quickbox
}

function _diskspaceupdate() {
  curl -o /srv/rutorrent/plugins/diskspace/action.php https://raw.githubusercontent.com/JMSDOnline/quick-box-update/master/v1-6/action.php >>"${OUTTO}" 2>&1
  chown -R www-data: /srv/rutorrent/plugins/diskspace
}

function _complete() {
  echo "Update process ${green}COMPLETE${normal}"
}

_intro
_bashrc
_diskspaceupdate
_complete

v1-3/quickbox-update-v1-3.sh
#!/bin/bash
#
# [Quick Box Update v1.3 Script]
#
# GitHub:   https://github.com/JMSDOnline/quick-box
# Author:   Jason Matthews
# URL:      https://jmsolodesigns.com/code-projects/quick-box/seedbox-installer
#
# 

HOSTNAME1=$(hostname -s);
VERSION="1.3"
OUTTO="/root/quick-box-update-v1-3.log"

#Script Console Colors
black=$(tput setaf 0);red=$(tput setaf 1);green=$(tput setaf 2);yellow=$(tput setaf 3);blue=$(tput setaf 4);magenta=$(tput setaf 5);cyan=$(tput setaf 6);white=$(tput setaf 7);on_red=$(tput setab 1);on_green=$(tput setab 2);on_yellow=$(tput setab 3);on_blue=$(tput setab 4);on_magenta=$(tput setab 5);on_cyan=$(tput setab 6);on_white=$(tput setab 7);bold=$(tput bold);dim=$(tput dim);underline=$(tput smul);reset_underline=$(tput rmul);standout=$(tput smso);reset_standout=$(tput rmso);normal=$(tput sgr0);alert=${white}${on_red};title=${standout};sub_title=${bold}${yellow};repo_title=${black}${on_green};

function _string() { perl -le 'print map {(a..z,A..Z,0..9)[rand 62] } 0..pop' 15 ; }

# Download packaged version 1.3 Quick Box
curl -LO https://raw.githubusercontent.com/JMSDOnline/quick-box-update/master/v1-3/changelog.md >>"${OUTTO}" 2>&1
wget https://github.com/JMSDOnline/quick-box-update/raw/master/v1-3/skel.tar >>"${OUTTO}" 2>&1
tar xf skel.tar -C /etc/skel

# Ask if user would like to update Plex
function _intro() {
  echo
  echo
  echo "${title} Welcome to the Quick Box update kit! ${normal}"
  echo " version: ${VERSION}"
  echo
  echo "Several changes had been made. Of these updates the largest"
  echo "commit to Quick Box was the enhanced feature of Plex."
  echo "Plex now comes as a Public facing function - meaning there"
  echo "is no longer a need to figure out complicated Tunneling via ssh."
  echo
  echo "If you do not have Plex installed - nor care do you care to"
  echo "run the Plex update, you may simply select (${red}N${normal}) when the"
  echo "question presents itself."
  echo
  echo "The rest of the updates will run silently. You can review the update log"
  echo "at ${OUTTO}."
  echo
  echo "Let's review real quick some of the changes that version 1.3"
  echo "brings in."
  echo
  echo
  echo -n "${bold}${yellow}Are you ready to run the v1.3 update?${normal} (${bold}${green}Y${normal}/n): "
  read responce
  case $responce in
    [yY] | [yY][Ee][Ss] | "" ) update=yes ;;
    [nN] | [nN][Oo] ) update=no ;;
  esac
}

clear


# function to adjust apache permissions (30)
function _adjustapache() {
  sed -i 's/www-data     ALL=(ALL:ALL) NOPASSWD: \/usr\/bin\/quota, \/usr\/sbin\/repquota, \/usr\/bin\/reload, \/bin\/sed, \/usr\/bin\/pkill, \/usr\/bin\/killall/www-data     ALL=(ALL:ALL) NOPASSWD: ALL/g' /etc/sudoers
}

# function to ask for plexmediaserver (30)
function _askplex() {
  echo -ne "${yellow}Would you like to install and update Plex Media Server${normal} (Y/n): (Default: ${red}N${normal}) "; read responce
  case $responce in
    [yY] | [yY][Ee][Ss] )
    echo -n "Installing and Updating Plex ... "
      curl -o /srv/rutorrent/home/index.php.new https://raw.githubusercontent.com/JMSDOnline/quick-box-update/master/v1-3/index.php >>"${OUTTO}" 2>&1
      mv /srv/rutorrent/home/index.php /srv/rutorrent/home/index.php.v1.bak
      mv /srv/rutorrent/home/index.php.new /srv/rutorrent/home/index.php
      echo "ServerName ${HOSTNAME1}" | sudo tee /etc/apache2/conf-available/fqdn.conf >>"${OUTTO}" 2>&1
      sudo a2enconf fqdn >>"${OUTTO}" 2>&1
      touch /etc/apache2/sites-enabled/plex.conf
      chown www-data: /etc/apache2/sites-enabled/plex.conf
      echo "deb http://shell.ninthgate.se/packages/debian squeeze main" > /etc/apt/sources.list.d/plexmediaserver.list
      curl http://shell.ninthgate.se/packages/shell-ninthgate-se-keyring.key >>"${OUTTO}" 2>&1 | sudo apt-key add - >>"${OUTTO}" 2>&1
      apt-get update >>"${OUTTO}" 2>&1
      apt-get install -qq --yes --force-yes plexmediaserver >>"${OUTTO}" 2>&1
      service plexmediaserver restart >>"${OUTTO}" 2>&1
      echo "${OK}"
      ;;
    [nN] | [nN][Oo] | "") echo "${cyan}Skipping Plex install${normal} ... " ;;
    *) echo "${cyan}Skipping Plex install${normal} ... " ;;
  esac
}

function _bashrc() {
  curl -LO https://raw.githubusercontent.com/JMSDOnline/quick-box-update/master/v1-3/.bashrc >>"${OUTTO}" 2>&1
}

function _checkcss() {
  cd /srv/rutorrent/home/skins
  curl -o /srv/rutorrent/home/skins/quick.css.new https://raw.githubusercontent.com/JMSDOnline/quick-box-update/master/v1-3/quick.css >>"${OUTTO}" 2>&1
  mv /srv/rutorrent/home/skins/quirk.css /srv/rutorrent/home/skins/quirk.css.v1.bak
  mv /srv/rutorrent/home/skins/quick.css.new /srv/rutorrent/home/skins/quick.css
}

function _lshell() {
cat >/etc/lshell.conf<<'LS'
[global]
logpath         : /var/log/lshell/
loglevel        : 2

[default]
allowed         : ['cd','cp','-d','-dmS','git','irssi','ll','ls','-m','mkdir','mv','nano','pwd','-R','rm','rtorrent','rsync','-S','scp','screen','tar','unrar','unzip','nano','wget']
forbidden       : [';', '&', '|','`','>','<', '$(', '${','sudo','vi','vim','./']
warning_counter : 2
aliases         : {'ls':'ls --color=auto','ll':'ls -l'}
intro           : "== Seedbox Shell ==\nWelcome To Your Quick Box Seedbox Shell\nType '?' to get the list of allowed commands"
home_path       : '/home/%u'
env_path        : ':/usr/local/bin:/usr/sbin'
allowed_cmd_path: ['/home/']
scp             : 1
sftp            : 0
overssh         : ['ls', 'rsync','scp']
LS
}

function _complete() {
  for i in apache2 fail2ban vsftpd; do
    service $i restart >>"${OUTTO}" 2>&1
    systemctl enable $i >>"${OUTTO}" 2>&1
  done

  echo "Update process ${green}COMPLETE${normal}"
}

_intro
_askplex
_bashrc
_checkcss
_lshell
_complete